jupyter_server/services/sessions/sessionmanager.py

```diff
@@ -268,15 +286,51 @@ async def create_session(
         self._pending_sessions.remove(record)
         return result
 
-    def get_kernel_env(self, path):
-        """Return the environment variables that need to be set in the kernel"""
+    async def get_kernel_env(
+        self, path: Optional[str], name: Optional[ModelName] = None
+    ) -> Dict[str, str]:
+        """Return the environment variables that need to be set in the kernel
+
+        path : str
+            the path for the given session – seem to be a session id sometime.
+        name: str
+            Here the name is likely to be the name of the associated file
+            with the current kernel at startup time.
+
+
+        """
+        if name is not None:
+            session_dir = await ensure_async(self.contents_manager.get_kernel_path(path=path))
+            cwd = self.kernel_manager.cwd_for_path(path)
+            path = os.path.join(cwd, session_dir, name)
+        assert isinstance(path, str)
         return {**os.environ, "JPY_SESSION_NAME": path}
```

Comment:
I don't think `session_dir` is necessary here.  If the notebook file is in a subdirectory of `root_dir`, `JPY_SESSION_NAME` will contain a redundant directory name...
![Screen Shot 2022-11-30 at 12 34 26 PM](https://user-images.githubusercontent.com/22599560/204902818-4056b85d-2f9e-41d3-83c8-f5bcc7a78343.png)

Reply:
That is true, thanks.